### PR TITLE
Wrap retry in try/catch and emit `processError` on exception.

### PR DESCRIPTION
**Pull request by dubiousdavid**
This supersedes https://github.com/smartprocure/mongochangestream/pull/30

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -121,6 +122,11 @@ export interface CursorErrorEvent {
   error: CursorError
 }
 
+export interface ProcessErrorEvent {
+  type: 'processError'
+  error: unknown
```
Comment (daveyarwood) [inline src/types.ts:127]: Do we want to also include the context of whether the error occurred during the initial scan vs. while processing the change stream?

File: src/mongoChangeStream.test.ts [hunk 2]
```diff
@@ -224,6 +224,37 @@ describe('syncing', () => {
...
+    const options: SyncOptions = {
+      retry: {
+        retries: 2,
+        minTimeout: 500,
+      },
+    }
+    const sync = await getSync(options)
+    await initState(sync, db, coll)
+
+    let processError = false
```
Comment (daveyarwood) [inline src/mongoChangeStream.test.ts:238]: If we want to be extra explicit about the expected behavior, we could make this a count of processError events and assert that only one occurs.

**Review by daveyarwood — APPROVED** [review]
Nice, looks good!

I made some improvements to the README in #30 explaining how error handling works in mongochangestream. I think it would be good to pull those in. (The only thing that needs adjustment is that the event is called `processError` instead of just `error` like I had it in my PR.)

**Comment by dubiousdavid** [comment]
> Nice, looks good!
> 
> I made some improvements to the README in #30 explaining how error handling works in mongochangestream. I think it would be good to pull those in. (The only thing that needs adjustment is that the event is called `processError` instead of just `error` like I had it in my PR.)

Can you create a new PR with those changes?

File: src/mongoChangeStream.test.ts (hunk 2 above)
Comment (dubiousdavid) [reply src/mongoChangeStream.test.ts:238]: Sounds good.

File: src/types.ts (hunk 1 above)
Comment (dubiousdavid) [reply src/types.ts:127]: Good call!

**Comment by daveyarwood** [comment]
Yes, will do.

**Pull request merged**